src/rgb/linear.rs

```diff
@@ -44,84 +47,83 @@ pub struct Rgb<Wp = D65, T = f32>
     ///highest displayable amount.
     pub blue: T,
 
-    ///The white point associated with the color's illuminant and observer.
-    ///D65 for 2 degree observer is used by default.
-    pub white_point: PhantomData<Wp>,
+    ///The kind of RGB space. sRGB is the default.
+    pub space: PhantomData<S>,
 }
 
-impl<Wp, T> Copy for Rgb<Wp, T>
+impl<S, T> Copy for LinRgb<S, T>
     where T: Float,
-        Wp: WhitePoint<T>
+        S: RgbSpace,
 {}
 
-impl<Wp, T> Clone for Rgb<Wp, T>
+impl<S, T> Clone for LinRgb<S, T>
     where T: Float,
-        Wp: WhitePoint<T>
+        S: RgbSpace,
 {
-    fn clone(&self) -> Rgb<Wp, T> { *self }
+    fn clone(&self) -> LinRgb<S, T> { *self }
 }
 
-impl<T> Rgb<D65, T>
+impl<T> LinRgb<Srgb, T>
     where T: Float,
 {
     ///Linear RGB with white point D65.
-    pub fn new(red: T, green: T, blue: T) -> Rgb<D65, T> {
-        Rgb {
+    pub fn new(red: T, green: T, blue: T) -> LinRgb<Srgb, T> {
+        LinRgb {
             red: red,
             green: green,
             blue: blue,
-            white_point: PhantomData,
+            space: PhantomData,
         }
     }
 
     ///Linear RGB from 8 bit values with whtie point D65.
-    pub fn new_u8(red: u8, green: u8, blue: u8) -> Rgb<D65, T> {
-        Rgb {
+    pub fn new_u8(red: u8, green: u8, blue: u8) -> LinRgb<Srgb, T> {
+        LinRgb {
             red: flt::<T,_>(red) / flt(255.0),
             green: flt::<T,_>(green) / flt(255.0),
             blue: flt::<T,_>(blue) / flt(255.0),
-            white_point: PhantomData,
+            space: PhantomData,
         }
     }
 }
 
-impl<Wp, T> Rgb<Wp, T>
+impl<S, T> LinRgb<S, T>
     where T: Float,
-        Wp: WhitePoint<T>
+        S: RgbSpace,
 {
     ///Linear RGB.
-    pub fn with_wp(red: T, green: T, blue: T) -> Rgb<Wp, T> {
-        Rgb {
+    pub fn with_wp(red: T, green: T, blue: T) -> LinRgb<S, T> {
```

Comment:
Bit of a nitpick. Needs a better name than ```with_wp``` that we can use for all colors. Maybe something like ```with_color``` or ``` with_type```.